src/storage/WrappedExpiringResourceLocker.ts

```diff
@@ -0,0 +1,69 @@
+import { EventEmitter } from 'events';
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { getLoggerFor } from '../logging/LogUtil';
+import type { ExpiringLock } from './ExpiringLock';
+import type { ExpiringResourceLocker } from './ExpiringResourceLocker';
+import type { ResourceLocker } from './ResourceLocker';
+
+/**
+ * Allows the locking of resources which is needed for non-atomic {@link ResourceStore}s.
+ * Differs from {@Link ResourceLocker} by adding expiration logic.
+ */
+export class WrappedExpiringResourceLocker implements ExpiringResourceLocker {
+  protected readonly logger = getLoggerFor(this);
+
+  protected readonly locker: ResourceLocker;
+  protected readonly readTimeout: number;
+
+  /**
+   *
+   * @param locker - Instance of ResourceLocker to use for acquiring a lock.
+   * @param readTimeout - Time in ms after which reading a representation times out, causing the lock to be released.
+   */
+  public constructor(locker: ResourceLocker, readTimeout: number) {
+    this.locker = locker;
+    this.readTimeout = readTimeout;
+  }
+
+  /**
+   * Lock the given resource with a lock providing expiration functionality.
+   * @param identifier - Identifier of the resource that needs to be locked.
+   *
+   * @returns A promise containing the expiring lock on the resource.
+   */
+  public async acquire(identifier: ResourceIdentifier): Promise<ExpiringLock> {
+    const lock = await this.locker.acquire(identifier) as unknown as ExpiringLock;
+
+    const emitter = new EventEmitter();
+
+    // Implement the emit and on method by calling those functions on an EventEmitter object for this lock.
+    lock.emit = (event: string | symbol, ...args: any[]): boolean => emitter.emit(event, args);
+    lock.on = (event: string | symbol, listener: (...args: any[]) => void): ExpiringLock => {
+      emitter.on(event, listener);
+      return lock;
+    };
```

Comment:
Mmm we should not monkey patch an object.

Let's instead define (in this file is fine) a `class WrappedExpiringLock implements ExpiringLock extends EventEmitter` with a `constructor(innerLock: lock, timeout: number)`, and put the logic in there.

Then the `acquire` method of `WrappedExpiringResourceLocker`:
- creates the `innerLock` (as done now)
- creates the  `WrappedExpiringLock`

Reply:
`WrappedExpiringLock` doesn't need to extend `EventEmitter` since `ExpiringLock` already extends it. But I agree with everything else.

Reply:
Nevermind, it's extend so it is necessary 😄